consensus/src/dag/reliable_broadcast.rs

```diff
@@ -67,3 +84,108 @@ impl ReliableBroadcast {
         }
     }
 }
+
+#[derive(ThisError, Debug)]
+pub enum NodeBroadcastHandleError {
+    #[error("invalid parent in node")]
+    InvalidParent,
+    #[error("missing parents")]
+    MissingParents,
+    #[error("parents do not meet quorum voting power")]
+    NotEnoughParents,
+}
+
+pub struct NodeBroadcastHandler {
+    dag: Arc<RwLock<Dag>>,
+    signatures_by_round_peer: BTreeMap<Round, BTreeMap<Author, NodeDigestSignature>>,
+    signer: ValidatorSigner,
+    verifier: ValidatorVerifier,
+}
+
+impl NodeBroadcastHandler {
+    pub fn new(
+        dag: Arc<RwLock<Dag>>,
+        signer: ValidatorSigner,
+        verifier: ValidatorVerifier,
+    ) -> Self {
+        Self {
+            dag,
+            signatures_by_round_peer: BTreeMap::new(),
+            signer,
+            verifier,
+        }
+    }
+
+    pub fn gc_before_round(&mut self, min_round: Round) {
+        self.signatures_by_round_peer.retain(|r, _| r >= &min_round);
+    }
+
+    fn validate(&self, node: &Node) -> anyhow::Result<()> {
```

Comment:
we should add a `fn verify(self, verifier: &ValidatorVerifier) -> Result<Self>` to TDagMessage for all self contained verification